docs/src/dev-guide/coding-guidelines.md

```
@@ -53,6 +53,154 @@ To avoid including a state variable in a React Hook's dependency array, you can
+### Type definitions
+
+Split store types into three interfaces:
+
+* `{Name}Values` - state variables
+* `{Name}Actions` - action functions
+* `{Name}State` - union of values and actions
+
+When store gets too large, we slice `{Name}State` to `{Name}{Feature}Slice`.
+Then we unionize these slices to `{Name}State`.
+
+
```

markdownlint-cli2 (0.17.2)

Multiple consecutive blank lines
Expected: 1; Actual: 2

(MD012, no-multiple-blanks)